src/xcsync/SyncContext.cs

```diff
@@ -453,6 +453,22 @@ async Task SyncFromXcodeAsync (CancellationToken token)
 	{
 		Logger.Information (Strings.Sync.HeaderInformation, TargetDir, ProjectPath);
 
+		// repeated from SyncToXcodeAsync --> refactor to helper method?
```

Comment:
We should not be copying any file/directory with arbitrary extensions. We should always be using the project file to determine the files/directories that should be synced.
Additionally, the [XcsyncWorkspace](https://vscode.dev/github/dotnet/xcsync/blob/dev/mcumming/public-ci/src/xcsync/Projects/XcodeWorkspace.cs) maintains a list of files that should be synced in the [Items](https://vscode.dev/github/dotnet/xcsync/blob/dev/mcumming/public-ci/src/xcsync/Projects/XcodeWorkspace.cs#L47-L48) property, that property is initialized in the [LoadSyncableItems](https://vscode.dev/github/dotnet/xcsync/blob/dev/mcumming/public-ci/src/xcsync/Projects/XcodeWorkspace.cs#L115-L132) method. Those items are then processed [here](https://vscode.dev/github/dotnet/xcsync/blob/dev/mcumming/public-ci/src/xcsync/SyncContext.cs#L467-L473). 

What is missing is tracking of the `.plist` and `.xib` file extensions, and then handling the `SyncableContent` case.

Reply:
can't access the vscode dev links:

![Screenshot 2024-10-02 at 9 37 03 AM](https://github.com/user-attachments/assets/1e121b21-bedb-46de-9db8-531533b717a3)

`Scripts.GetFiles` does utilize the project file to identify the relevant sync items here: https://github.com/dotnet/xcsync/blob/0344c637529976873f615279dd92565a1a72cbc3/src/xcsync/Scripts.cs#L117

the current approach isn't tracking the changes successfully back, but I'll look into your suggestions to see what I can do here